aptos-indexer-processors-sdk/testing-framework/src/new_test_context.rs

```diff
@@ -0,0 +1,216 @@
+use crate::mock_grpc::MockGrpcServer;
+use anyhow::Context;
+use aptos_indexer_processor_sdk::{
+    aptos_indexer_transaction_stream::TransactionStreamConfig,
+    traits::processor_trait::ProcessorTrait,
+};
+use aptos_protos::{indexer::v1::TransactionsResponse, transaction::v1::Transaction};
+use serde_json::{to_string_pretty, Error as SerdeError};
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+use tokio::{
+    sync::Mutex,
+    time::{self, Duration as TokioDuration},
+};
+use tokio_retry::{
+    strategy::{jitter, ExponentialBackoff},
+    Retry,
+};
+use url::Url;
+
+const INDEXER_GRPC_DATA_SERVICE_URL: &str = "http://localhost:51254";
+
+pub struct SdkTestContext {
+    pub transaction_batches: Vec<Transaction>,
+}
+
+impl SdkTestContext {
+    pub async fn new(txn_bytes: &[&[u8]]) -> anyhow::Result<Self> {
+        let transaction_batches = txn_bytes
+            .iter()
+            .enumerate()
+            .map(|(idx, txn)| {
+                serde_json::from_slice::<Transaction>(txn).map_err(|err| {
+                    anyhow::anyhow!(
+                        "Failed to parse transaction at index {}: {}",
+                        idx,
+                        format_serde_error(err)
+                    )
+                })
+            })
+            .collect::<Result<Vec<Transaction>, _>>()?;
+
+        let context = SdkTestContext {
+            transaction_batches,
+        };
+
+        // Create mock GRPC transactions and setup the server
+        let transactions = context.transaction_batches.clone();
+        let transactions_response = vec![TransactionsResponse {
+            transactions,
+            ..TransactionsResponse::default()
+        }];
+
+        // Call setup_mock_grpc to start the server and get the port
+        context.setup_mock_grpc(transactions_response, 1).await;
+
+        Ok(context)
+    }
+
+    /// Run the processor and pass user-defined validation logic
+    pub async fn run<F>(
+        &mut self,
+        processor: &impl ProcessorTrait,
+        db_url: &str,
```

Comment:
It looks like this `db_url` is only used to be passed into `verification_f`. My question is why does this `run` function need the `db_url` variable? The test could refer to the `db_url` inside of the function instead of having it passed down from another function.

Reply:
I left a comment about this, I agreed that we wouldn't need to pass in db_url here. as verification function can directly access db_url